test/run-test.sh

```diff
@@ -105,11 +112,18 @@ if ! build "${tag}" ${build_arg}; then
     FAIL "build failed unexpectedly"
 fi
 
-if ! docker run --privileged "iptables-wrapper-test-${tag}" /bin/sh ${dash_x:-} /test.sh legacy; then
-    FAIL "failed legacy iptables / new rules test"
+if ! docker run --privileged "iptables-wrapper-test-${tag}" /tests -test.v -test.run "^TestIPTablesWrapperLegacy$" ; then
+    FAIL "failed legacy iptables"
+fi
+if ! docker run --privileged "iptables-wrapper-test-${tag}" /tests -test.v -test.run "^TestIPTablesWrapperNFT$" ; then
+    FAIL "failed nft iptables"
+fi
```

Comment:
Is there some reason for running them separately like this rather than just doing a single `docker run` for the whole test suite?

Reply:
Yeah, running one test modifies the container (the symlinks are altered to point to a particular mode).

So instead of building some kind of cleanup in the tests that try to undo all the changes and restore the initial state, I just run them in different containers. It seemed simpler.